oximeter/db/src/client/mod.rs

```diff
@@ -1437,9 +1437,34 @@ impl Client {
         if replicated {
             sql.push_str(" AND engine = 'ReplicatedMergeTree'");
         }
-        self.execute_with_body(sql).await.map(|(_summary, body)| {
-            body.lines().map(ToString::to_string).collect()
-        })
```

Comment:
I'm a little confused - is the usage of `execute_with_body` the reason this was using the HTTP interface? There seem to be many some other invocations of this function -- see, `select_matching_timeseries_info`.

Can we (should we?) remove those too?

Reply:
That's correct. I'm doing this in chunks, to avoid a 10K line PR :) Moving to the native client for selecting data is #6886

Reply:
For some more context, I'm doing this ship of Theseus style. I've added the native client, then I'm taking chunks of database operations and replacing them one at a time. Then I'll remove the HTTP client. We're nearly there, just need get #7013 done and a PR closing #6886!